### PR TITLE
Migrate rokt-audiences to new authentication method

### PR DESCRIPTION
**Pull request by GoodmanBen**
## Testing
- [x] Updated [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing). Successfully checked authentication and test event submission:
image
image
- [x] [If destination is already live] Tested for backward compatibility of destination. **Note:** New required fields are a breaking change.  --> This integration is in private beta and we have established a migration plan with the one active private beta user.
- [ ] [Segmenters] Tested in the staging environment
- [ ] [Segmenters] [If applicable for this change] Tested for regression with Hadron.

### CODE REVIEW

**Comment by joe-ayoub-segment** [comment]
hi @GoodmanBen thanks for raising this draft PR. 
I won't review it until you ask me to - however I had a quick scan and noticed that all of the auth settings are changing. 
This will be a breaking change, and will prevent any data collection for customers who are already using the Integration. 
Customers will need to reconfigure auth settings before data collection to Rokt can resume. 
Just want to make sure you are aware of this.

**Comment by GoodmanBen** [comment]
> hi @GoodmanBen thanks for raising this draft PR. I won't review it until you ask me to - however I had a quick scan and noticed that all of the auth settings are changing. This will be a breaking change, and will prevent any data collection for customers who are already using the Integration. Customers will need to reconfigure auth settings before data collection to Rokt can resume. Just want to make sure you are aware of this.

Hi @joe-ayoub-segment thank you, this has been tested and is ready for review. Yes, thank you for flagging. We understand that this is a breaking change. This extension is currently in private beta and we have coordinated with the one user on how to transition to the new authentication approach.

File: packages/destination-actions/src/destinations/rokt-audiences/upsertCustomAudiences/index.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@ const action: ActionDefinition<Settings, Payload> = {
         '@if': {
           exists: { '@path': '$.context.traits.email' },
           then: { '@path': '$.context.traits.email' },
-          else: { '@path': '$.traits.email' }
+          else: { '@path': '$.email' }
```
Comment (joe-ayoub-segment) [inline packages/destination-actions/src/destinations/rokt-audiences/upsertCustomAudiences/index.ts:40]:
```suggestion
          else: { '@path': '$.traits.email' }
```

Comment (joe-ayoub-segment) [reply packages/destination-actions/src/destinations/rokt-audiences/upsertCustomAudiences/index.ts:40]: $.traits.email is the the location where the email address will be when Engage outputs an identify() call from the Audience. 
$.context.traits.email is the the location where the email address will be when Engage outputs an track() call from the Audience.

**Review by joe-ayoub-segment — DISMISSED** [review]

File: packages/destination-actions/src/destinations/rokt-audiences/upsertCustomAudiences/index.ts (hunk 1 above)
Comment (GoodmanBen) [reply packages/destination-actions/src/destinations/rokt-audiences/upsertCustomAudiences/index.ts:40]: Ah got it, accepting suggestion then merging!

**Review by joe-ayoub-segment — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
We're in a deploy freeze so next deploy scheduled for 3-Dec

**Pull request merged**